lib/inspector.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const connect = process.binding('inspector').connect;
+const EventEmitter = require('events');
+const util = require('util');
+
+if (!connect)
+  throw new Error('Inspector is not available');
+
+const connectionSymbol = Symbol('connectionProperty');
+const messageCallbacksSymbol = Symbol('messageCallbacks');
+const nextIdSymbol = Symbol('nextId');
+const onMessageSymbol = Symbol('onMessage');
+
+class Session extends EventEmitter {
+  constructor() {
+    super();
+    this[connectionSymbol] = null;
+    this[nextIdSymbol] = 1;
+    this[messageCallbacksSymbol] = new Map();
+  }
+
+  connect() {
+    if (this[connectionSymbol])
+      throw new Error('Already connected');
+    this[connectionSymbol] =
+        connect((message) => this[onMessageSymbol](message));
+  }
+
+  [onMessageSymbol](message) {
+    const parsed = JSON.parse(message);
+    if (parsed.id) {
+      const callback = this[messageCallbacksSymbol].get(parsed.id);
+      this[messageCallbacksSymbol].delete(parsed.id);
+      if (callback)
+        callback(parsed.error || null, parsed.result || null);
+    } else {
+      this.emit(parsed.method, parsed);
+      this.emit('inspectorNotification', parsed);
+    }
+  }
+
+  post(method, params, callback) {
+    if (typeof method !== 'string')
+      throw new TypeError(
+          `"method" must be a string, got ${typeof method} instead`);
+    if (!callback && util.isFunction(params)) {
+      callback = params;
+      params = null;
+    }
+    if (params && typeof params !== 'object')
+      throw new TypeError(
+          `"params" must be an object, got ${typeof params} instead`);
+    if (callback && typeof callback !== 'function')
+      throw new TypeError(
+          `"callback" must be a function, got ${typeof callback} instead`);
+
+    if (!this[connectionSymbol])
+      throw new Error('Session is not connected');
+    const id = this[nextIdSymbol]++;
+    const message = {id, method};
+    if (params) {
+      message['params'] = params;
+    }
+    if (callback) {
+      this[messageCallbacksSymbol].set(id, callback);
+    }
+    this[connectionSymbol].dispatch(JSON.stringify(message));
+  }
+
+  disconnect() {
+    if (!this[connectionSymbol])
+      return;
+    this[connectionSymbol].disconnect();
+    this[connectionSymbol] = null;
+    const remainingCallbacks = this[messageCallbacksSymbol].values();
+    var error = new Error('Session was closed');
+    for (const callback of remainingCallbacks) {
+      callback(error);
```

Comment:
Consider doing `for (const callback of remainingCallbacks) process.nextTick(callback, new Error(...))` for two reasons:

1. It ensures subsequent callbacks still run if a callback throws, and
2. `new Error(...)` gives each callback its own copy.